src/replication_test.c

```diff
@@ -197,48 +192,94 @@ send_io_resp(int fd, zvol_io_hdr_t *io_hdr, void *buf)
 	return 0;
 
 }
+
+static void
+usage(void)
+{
+	printf("replica_test [options]\n");
+	printf("options:\n");
+	printf(" -i target ip\n");
+	printf(" -p target port\n");
+	printf(" -I replica ip\n");
+	printf(" -P replica port\n");
+	printf(" -V volume path\n");
+	printf(" -t timeout in seconds\n");
+}
+
+
 int
 main(int argc, char **argv)
 {
-	if(argc < 5) {
-		exit(EXIT_FAILURE);
-	}
-	char *ctrl_ip = argv[1];
-	int ctrl_port = atoi(argv[2]);
-	char *replicaip = argv[3];
-	int replica_port = atoi(argv[4]);
-	char *test_vol = argv[5];
+	char ctrl_ip[MAX_IP_LEN];
+	int ctrl_port = 0;
+	char replica_ip[MAX_IP_LEN];
+	int replica_port = 0;
+	char test_vol[1024];
 	int sleeptime = 0;
 	struct zvol_io_rw_hdr *io_rw_hdr;
 	zvol_op_open_data_t *open_ptr;
-
-	if (argv[6] != NULL) {
-		printf("got 6th arg.. not setting healthy\n");
-		if_healthy = 0;
-//		sleeptime = atoi(argv[6]);
-	}
 	int iofd = -1, mgmtfd, sfd, rc, epfd, event_count, i;
 	int64_t count;
 	struct epoll_event event, *events;
 	uint8_t *data, *data_ptr_cpy;
 	uint64_t nbytes = 0;
-	int vol_fd = open(test_vol, O_RDWR, 0666);
+	int vol_fd;
 	zvol_op_code_t opcode;
-	zvol_io_hdr_t *io_hdr = malloc(sizeof(zvol_io_hdr_t));
-	zvol_io_hdr_t *mgmtio = malloc(sizeof(zvol_io_hdr_t));
-	
-	data = NULL;
+	zvol_io_hdr_t *io_hdr;
+	zvol_io_hdr_t *mgmtio;
 	bool read_rem_data = false;
 	bool read_rem_hdr = false;
 	uint64_t recv_len = 0;
 	uint64_t total_len = 0;
 	uint64_t io_hdr_len = sizeof(zvol_io_hdr_t);
+	int io_timeout = 0;
```

Comment:
instead of io_timeout, make it as io_cnt

Reply:
ok